test/multi-sign-test.js

```diff
@@ -0,0 +1,376 @@
+      mTx.setSequence(getAliceSequence() + 1);
+
+      let signer = mTx.multiSign(accounts.bob.account, accounts.bob.secret);
+      assert(signer.Account);
+      assert(signer.SigningPubKey);
+      assert(signer.TxnSignature);
+
+      tx.addMultiSigner(signer);
+
+      tx.once('submitted', function(res) {
+        assert.strictEqual(res.engine_result, 'temBAD_SIGNATURE');
+        callback();
+      });
+
+      tx.submit();
+    }
+    ], done);
+  });
+
+  test('Invalid signature -- malformed signer', function(done) {
+    async.series([
+    function(callback) {
+      let tx = $.remote.createTransaction('SignerListSet', {
+        account: accounts.alice.account,
+        signers: [
+          {account: accounts.bob.account, weight: 1},
+        ],
+        signerQuorum: 1
+      });
+
+      testutils.submit_transaction(tx, callback);
+    },
+
+    function(callback) {
+      let tx = $.remote.createTransaction('AccountSet', {account: accounts.alice.account});
+      tx.setFee(100);
+      tx.setSequence(getAliceSequence());
+      tx.setLastLedgerSequenceOffset(5);
+
+      let mTx = Transaction.from_json(tx.getMultiSigningJson());
+
+      let signer = mTx.multiSign(accounts.bob.account, accounts.bob.secret);
+      // Tamper with signer after multi-signing
+      signer.Account = accounts.carol.account;
+
+      assert(signer.Account);
+      assert(signer.SigningPubKey);
+      assert(signer.TxnSignature);
+
+      tx.addMultiSigner(signer);
+
+      tx.once('submitted', function(res) {
+        assert.strictEqual(res.engine_result, 'temBAD_SIGNATURE');
+        callback();
+      });
+
+      tx.submit();
+    }
+    ], done);
+  });
+
+  test('Invalid signature -- SigningPubKey non-empty', function(done) {
+    async.series([
+    function(callback) {
+      let tx = $.remote.createTransaction('SignerListSet', {
+        account: accounts.alice.account,
+        signers: [
+          {account: accounts.bob.account, weight: 1},
+        ],
+        signerQuorum: 1
+      });
+
+      testutils.submit_transaction(tx, callback);
+    },
+
+    function(callback) {
+      let tx = $.remote.createTransaction('AccountSet', {account: accounts.alice.account});
+      tx.setFee(100);
+      tx.setSequence(getAliceSequence());
+      tx.setLastLedgerSequenceOffset(5);
+
+      let mTx = Transaction.from_json(tx.getMultiSigningJson());
+
+      let signer = mTx.multiSign(accounts.bob.account, accounts.bob.secret);
+      assert(signer.Account);
+      assert(signer.SigningPubKey);
+      assert(signer.TxnSignature);
+
+      tx.addMultiSigner(signer);
+
+      tx.once('presubmit', function(res) {
+        // SigningPubKey must be empty
+        tx.setSigningPubKey(tx.getSigningPubKey());
+      });
+      tx.once('submitted', function(res) {
+        assert(res.error);
+        assert.strictEqual(res.remote.error, 'invalidParams');
+        callback();
+      });
+
+      tx.submit();
```

Comment:
It seems `setup` sets local_signing to `false` for each `test` so that answers that question, and explains the `invalidParams`.